### PR TITLE
Remove reference to googlegroup

### PR DESCRIPTION
**Pull request by jlaehne**
Removes reference to google group based on the discussion at:
hyperspy/hyperspy#2252

[X] Remove googlegroup from `CONTRIBUTING`
[X] Add note on 'watching' GitHub releases to user guide

### CODE REVIEW

File: doc/user_guide/install.rst [hunk 1]
```diff
@@ -14,6 +14,12 @@ see the :ref:`anaconda-install` section.
 To enable context-menu (right-click) shortcut in a chosen folder, use the
 `start_jupyter_cm <https://github.com/hyperspy/start_jupyter_cm>`_ library.
 
+.. note::
+
+    If you want to be notified about new releases, please *Watch (Releases only)* 
+    the `hyperspy repositorium on GitHub <https://github.com/hyperspy/hyperspy/>`_
```
Comment (tjof2) [inline doc/user_guide/install.rst]: The standard term is "repository" I think.

Comment (jlaehne) [reply doc/user_guide/install.rst]: sure ;-)

**Review by tjof2 — CHANGES_REQUESTED** [review]
Needs to be based off RELEASE_next_patch I think @jlaehne

**Comment by tjof2** [comment]
Same problem as before though I'm afraid, it includes commits that shouldn't be there.

**Comment by jlaehne** [comment]
OK, figured it out should be fine now.

**Review by tjof2 — APPROVED** [review]

**Pull request merged**